### ISSUE TITLE
Crowdstrike quarantines SOME executable entrypoints upon installation on Windows

### ISSUE THREAD

**Issue by johannesloibl** [astral-sh/uv#7324]
Hey,

don't know if this is the right place, but my company recently installed Crowdstrike (yeah, that one)
and it is now messing with me after switching to awesome UV.

For only **some** of our internal Python packages, Crowdstrike is quarantining the EXE that is created by UV (because of defined application entrypoints of the library),
when i try installing it via ``uv tool install`` or creating a venv and installing the library using ``uv pip install``.

Why i'm creating this issue here? Well, if i'm using ``pip`` and ``pipx`` to install,
everything works fine.
Maybe Crowdstrike has a problem with the ``uv-trampoline`` bootstrap code? This seems to be the only difference between executables from UV and PIPX, or am i wrong?

![image](https://github.com/user-attachments/assets/8187aaba-1c75-4e81-80bf-82b15b1b5148)

UV version: uv 0.4.9 (77d278f68 2024-09-10)

**Comment by johannesloibl** [comment]
@ofek FYI, your PyApps are also affected by this, just got the confirmation of a colleague.

**Comment by zanieb** [comment]
Thanks for the report.

Did you also report this as a false-positive to Crowdstrike? I'd love to hear what they say too. We're not doing anything malicious :)

**Comment by johannesloibl** [comment]
Yes our CyberSec team is reporting it to them, but i probably don't get any information i could share with you. Since they are doing a lot of scanning based on ML models, this can take some time until they updated to model to not flag the executables anymore, if they do this at all...

**Comment by magoocas** [comment]
Suffering from the same problem. Our cybersec team has also submitted a report on the false-positive to Crowdstrike. 

My workaround in the meantime is to use .cmd "shebangs" instead of the trampoline feature.

[cs_workaround.patch (I was on commit b4fa46bf)](https://github.com/user-attachments/files/17667601/cs_workaround.patch)

Works for my use case, YMMV!

**Comment by axel-kah** [comment]
Just wanted to report that this is still a blocking issue (in enterprise environments anyway) since crowdstrike has not adapted their scanners yet.
With `uv` being used under the hood of more and more tools, the impact is growing as well 😐
